src/main/java/bitflow4j/io/file/OutputFiles.java

```diff
@@ -0,0 +1,118 @@
+package bitflow4j.io.file;
+
+import bitflow4j.AbstractPipelineStep;
+import bitflow4j.Header;
+import bitflow4j.Sample;
+import bitflow4j.io.MarshallingSampleWriter;
+import bitflow4j.io.marshall.Marshaller;
+import bitflow4j.script.registry.Description;
+
+import java.io.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Created by kevinstyp on 15/04/2019.
+ */
+@Description("Uses the provided template in parameter file to save samples in their respective file. Example call: " +
+        "output-files('./${data_type}/${serial}.csv', 'CSV') " +
+        "This will evaluate the tags data_type and serial of every sample and save the sample in the proper file.")
+public class OutputFiles extends AbstractPipelineStep {
```

Comment:
The output-files step is used when the user expects many different files to be created. The current implementation opens a new file stream whenever the target file of a sample changes. This is not compatible with the Go implementation and will lead to a large number of files with different suffixes (xxx-1.csv, xxx-2.csv, ...)
The OutputFiles step should use a ForkStep (probably inherit from Fork, not sure). The forking should happen in the same way like in ForkTemplate, but each sub pipeline should only contain a single FileSink, which is constructed based on the resolved tag template.

Reply:
I know this means that the current implementation has to be changed substantially, but I hope this is doable :)